packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -820,3 +820,89 @@ export const parseSelectedSections = (
 
   return selectedSections;
 };
+
+export const getSectionValueText = <TDate extends PickerValidDate>(
+  section: FieldSection,
+  utils: MuiPickersAdapter<TDate>,
+  currentDate: TDate | null,
+): string | undefined => {
+  if (!section.value) {
+    return undefined;
+  }
+  switch (section.type) {
+    case 'month': {
+      if (section.contentType === 'digit') {
+        return currentDate && utils.isValid(currentDate)
+          ? utils.format(currentDate, 'month')
+          : utils.format(utils.setMonth(utils.date()!, Number(section.value) - 1), 'month');
+      }
+      const parsedDate = utils.parse(section.value, section.format);
+      return parsedDate ? utils.format(parsedDate, 'month') : undefined;
+    }
+    case 'day':
+      return section.contentType === 'digit'
+        ? utils.format(
+            utils.setDate(utils.startOfYear(utils.date()!), Number(section.value)),
+            'dayOfMonthFull',
+          )
+        : section.value;
+    case 'meridiem':
+      return currentDate && utils.isValid(currentDate)
+        ? utils.format(currentDate, 'meridiem')
+        : undefined;
+    case 'weekDay':
+      return currentDate && utils.isValid(currentDate)
+        ? utils.format(currentDate, 'weekday')
+        : undefined;
+    default:
+      return undefined;
+  }
+};
+
+export const getSectionValueNow = <TDate extends PickerValidDate>(
+  section: FieldSection,
+  utils: MuiPickersAdapter<TDate>,
+  currentDate: TDate | null,
+): number | undefined => {
+  if (!section.value || currentDate === null) {
+    return undefined;
+  }
+  switch (section.type) {
+    case 'weekDay': {
+      if (section.contentType === 'letter') {
+        // TODO: consider handling case when `currentDate` is not valid
```

Comment:
If you have ideas on how to resolve the weekday number from string like `Mon` or `Monday` - I'm very open to suggestions. 🙈